core/src/CoreService.ts

```diff
@@ -67,11 +67,31 @@ export module Services.Core {
      * @returns {*}
      */
     public exports(): any {
+      let exportedMethods: any = {};
+      if(process.env["sails_redbox__mochaTesting"] === "true") {
+        const allProperties = [
+          ...Object.getOwnPropertyNames(Object.getPrototypeOf(this)), // Prototype methods
+          ...Object.getOwnPropertyNames(this), // Instance properties
+        ];
+        
+        
+        const uniqueProperties = Array.from(new Set(allProperties));
+        uniqueProperties.forEach((property) => {
+          const value = (this as any)[property];
+    
+          // Check if the property is a function
+          if (typeof value === "function" && property !== "constructor") {
+            exportedMethods[property] = value.bind(this); // Bind the method to maintain `this` context
+          }
+        
+        });
+        console.error("Exported Methods for Mocha Testing: ", exportedMethods);
+      } else {
```

Comment:
This is interesting. Looks useful.
Is the only intent for this to be able to test private methods / functions, or are there other uses?

Reply:
Only for testing private methods